### PR TITLE
Feature/add_Vietnamese_Telex

### PR DESCRIPTION
**Pull request by chihung93**
Hi @tibbi , I have finished the support of Vietnamese - Telex

Feature:

1. Added a Vietnam Telex in the Setting screen
2. Added the telex extension (Reference: https://github.com/florisboard/florisboard/blob/aa00eb6e79081fb602c68e55b7d5e80bfe2ae322/app/src/main/assets/ime/keyboard/org.florisboard.composers/extension.json#L17)
3. Using telex in Edit text.
Thanks.

Demo:

[Demo_telex.webm](https://github.com/SimpleMobileTools/Simple-Keyboard/assets/5199109/f6e7280d-4535-42d9-9da0-f2617c6a2036)

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/keyboard/activities/SettingsActivity.kt [hunk 1]
```diff
@@ -14,6 +14,7 @@ import com.simplemobiletools.keyboard.extensions.getKeyboardLanguages
 import com.simplemobiletools.keyboard.helpers.KEYBOARD_HEIGHT_MULTIPLIER_LARGE
 import com.simplemobiletools.keyboard.helpers.KEYBOARD_HEIGHT_MULTIPLIER_MEDIUM
 import com.simplemobiletools.keyboard.helpers.KEYBOARD_HEIGHT_MULTIPLIER_SMALL
+import com.simplemobiletools.keyboard.helpers.LANGUAGE_VIETNAMESE_TELEX
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/activities/SettingsActivity.kt:17]: guess this import isnt needed anymore

File: app/src/main/kotlin/com/simplemobiletools/keyboard/helpers/EmojiHelper.kt [hunk 2]
```diff
@@ -59,3 +62,27 @@ fun parseRawEmojiSpecsFile(context: Context, path: String): MutableList<String>
...
 }
+
+
+fun parseRawJsonSpecsFile(context: Context, path: String): HashMap<String,String> {
+    if (cachedVNTelexData.isNotEmpty()) {
+        return cachedVNTelexData
+    }
+
+    try {
+        val  inputStream: InputStream = context.assets.open(path)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/helpers/EmojiHelper.kt:73]: use ctrl + alt + L to format text, also avoid writing double empty lines like amove

File: app/src/main/kotlin/com/simplemobiletools/keyboard/services/SimpleKeyboardIME.kt [hunk 3]
```diff
@@ -178,10 +178,38 @@ class SimpleKeyboardIME : InputMethodService(), OnKeyboardActionListener, Shared
...
+                        }
+                        else -> {
+                            inputConnection.commitText(codeChar.toString(), 1)
+                            if (originalText == null) {
+                                updateShiftKeyState()
+                            }
+                        }
                     }
+
+
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/services/SimpleKeyboardIME.kt:212]: remove these empty lines too

File: app/src/main/kotlin/com/simplemobiletools/keyboard/views/MyKeyboardView.kt [hunk 4]
```diff
@@ -418,6 +418,11 @@ class MyKeyboardView @JvmOverloads constructor(context: Context, attrs: Attribut
         }
 
         setupEmojiPalette(toolbarColor = toolbarColor, backgroundColor = mBackgroundColor, textColor = mTextColor)
+        if(context.config.keyboardLanguage == LANGUAGE_VIETNAMESE_TELEX){
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/views/MyKeyboardView.kt:421]: again, format code to have proper spacing

**Comment by chihung93** [comment]
Thanks for your feedback. I have fixed all issues. @tibbi

**Comment by tibbi** [comment]
looks great, thanks :)

**Comment by tranhongs0n** [comment]
@chihung93, @tibbi Thank you all for adding Vietnamese to this soft app, but there are still many bugs and I almost can't use this keyboard to type Vietnamese. I hope everyone will consider testing it more thoroughly.

**Comment by tibbi** [comment]
ok, will try improving it once we find someone that has experience with it :)

**Comment by dmitriy-chernysh** [comment]
I see Vietnamese key layout (xml/keys_letters_....xml) is missing in the app

**Pull request merged**